crates/egui/src/ui.rs

```diff
@@ -329,20 +330,44 @@ impl Ui {
         self.placer.layout()
     }
 
-    /// Should text wrap in this [`Ui`]?
+    /// Which wrap mode should the text use in this [`Ui`]?
     ///
-    /// This is determined first by [`Style::wrap`], and then by the layout of this [`Ui`].
-    pub fn wrap_text(&self) -> bool {
+    /// This is determined first by [`Style::wrap_mode`], and then by the layout of this [`Ui`].
+    pub fn wrap_mode(&self) -> TextWrapMode {
+        // `wrap` handling for backward compatibility
+        #[allow(deprecated)]
         if let Some(wrap) = self.style.wrap {
```

Comment:
We should start with the new (`wrap_mode`) and only use the deprecated items as the fallback